src/mscorlib/src/System/Globalization/TextInfo.cs

```diff
@@ -482,6 +482,15 @@ public unsafe String ToTitleCase(String str)
                 charType = CharUnicodeInfo.InternalGetUnicodeCategory(str, i, out charLen);
                 if (Char.CheckLetter(charType))
                 {
+                    // Special case to check for Dutch specific titlecasing with
+                    // "IJ" characters at the beginning of word
+                    if (IsDutchCulture(str) && IsIjAtCurrentPosition(str, i))
```

Comment:
>IsDutchCulture(str) [](start = 24, length = 19)

this can be cached in a local variable instead of calling the same method again inside the loop